mantidproject/mantid
# Create reflectometry library

**Description of work.**
A reflectometry-specific algorithm library has been created in `Mantid/Framework`, similar to `Mantid/Framework/Muon`, as part of the workflow algorithm history fix umbrella issue #23012.

I have scoured the documentation for every reflectometry-specific algorithm I could find and picked out a couple of others by hand. **Note:** this work may not include all of the undocumented ILL reflectometry algorithms so it may be worth someone from that facility checking, either raising any missed in this PR or moving them in a subsequent PR.

This PR also includes the following changes:
- `BoxControllerSettingsAlgorithm` has been moved from `MDAlgorithms` to `API` so that it is still accessible by `ConvertToReflectometryQ`. The tests have been simplified slightly to remove dependence on DataHandling.
- Add a missing include of `Algorithm.tcc` to `GetEiMonDet3.cpp`.
- Minor changes to some tests to ensure framework is initialized and to create child algorithms in a more consistent way.

**To test:**
Check that the tests pass and maybe try running a couple of the algorithms.

Refs #23012

*This does not require release notes* because **it is an internal change**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
